gum/backend-darwin/gumdarwinsymbolicator.c

```diff
@@ -314,7 +340,8 @@ gum_darwin_symbolicator_details_from_address (GumDarwinSymbolicator * self,
   symbol = CSSymbolicatorGetSymbolWithAddressAtTime (self->handle, address,
       kCSNow);
   if (CSIsNull (symbol))
-    return FALSE;
+    return gum_darwin_symbolicator_objc_details_from_address (self,
```

Comment:
i think this behaviour should be exposed in some way as a configuration option. 

The reason is that the objc api resolver (used under the hood by this feature) can't be used safely if the app is still paused in early instrumentation, and making this fallback by default may introduce stability issues to the existing scripts which use the `DebugSymbol` API during early instrumentation.

Reply:
Makes sense, how do you suggest we pass in or read the config data though?

Reply:
Probably with a settings object, i think @oleavr could point us to good examples

Reply:
Actually, wouldn't resolver be `NULL` in case that `dlopen()` fails for `libobjc.A.dylib`? So we just need to return `FALSE` in that case?

Reply:
the real issue isn't about `dlopen()` failing, it's described better here: https://github.com/frida/frida/issues/1167#issuecomment-707284109

Reply:
I wonder if we could somehow avoid these side-effects by detecting that a class isn't initialized, making the ObjC ApiResolver safe in all scenarios. Adding a configuration option would work, but I'm hopeful there's a way we can avoid it.

Reply:
After discussing with @oleavr, we'll leave this as-is for now since we can't reproduce it at all.